### PR TITLE
chore: active filter pills improvements

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Enhancement**

### Quick demo links
* [Partner artworks](https://active-filter-pills.artsy.net/partner/artsy-2/works)
* [Artist artworks](https://active-filter-pills.artsy.net/artist/banksy/works-for-sale)

### Description
Small improvements that fix the offset problem on Partner Artworks page when there are no active filters
#### Before
before

#### After
after

Also these changes allow us to customize the display of active filter pills in more detail and "Create Alert" button and simplify the addition of active filter pills to other grids

### CODE REVIEW

File: src/v2/Components/ArtworkFilter/index.tsx [hunk 1]
```diff
@@ -104,8 +102,7 @@ export const BaseArtworkFilter: React.FC<
   relayRefetchInputVariables = {},
   relayVariables = {},
   viewer,
-  FilterPills,
-  CreateAlertButton,
+  ContentAboveArtworks,
```
Comment (dimatretyak) [inline src/v2/Components/ArtworkFilter/index.tsx:105]: **question**: is such a prop name suitable or is it better to rename it?

Comment (dimatretyak) [reply src/v2/Components/ArtworkFilter/index.tsx:105]: I like this name 👌

File: src/v2/Components/SavedSearchAlert/useActiveFilterPills.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  initialArtworkFilterState,
+  useArtworkFilterContext,
+} from "v2/Components/ArtworkFilter/ArtworkFilterContext"
+import { DEFAULT_METRIC } from "v2/Components/ArtworkFilter/Utils/metrics"
+import { usePrepareFiltersForPills } from "v2/Components/ArtworkFilter/Utils/usePrepareFiltersForPills"
+import { FilterPill } from "./types"
+import { extractPillsFromCriteria } from "./Utils/extractPills"
+import { getAllowedSearchCriteria } from "./Utils/savedSearchCriteria"
+
+export const useActiveFilterPills = (defaultPills: FilterPill[] = []) => {
```
Comment (gkartalis) [inline src/v2/Components/SavedSearchAlert/useActiveFilterPills.ts:13]: 👏👏👏

File: src/v2/Components/ArtworkFilter/index.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/v2/Components/ArtworkFilter/index.tsx:105]: Done

**Review by gkartalis — APPROVED** [review]
🚀 👏 Great refactor!  🚢 🚢

**Pull request merged**